auth-api/src/main/java/io/stargate/auth/api/AuthResource.java

```diff
@@ -87,29 +90,83 @@ public Response createToken(Secret body) {
   @Path("/auth")
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Response createToken(Credentials body) {
-    if (body == null) {
+  public Response createToken(Credentials credentials) {
+    if (credentials == null) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide a body to the request"))
           .build();
     }
 
-    if (body.getUsername() == null || body.getUsername().equals("")) {
+    if (credentials.getUsername() == null || credentials.getUsername().equals("")) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide username in request"))
           .build();
     }
 
-    if (body.getPassword() == null || body.getPassword().equals("")) {
+    if (credentials.getPassword() == null || credentials.getPassword().equals("")) {
       return Response.status(Response.Status.BAD_REQUEST)
           .entity(new Error("Must provide password in request"))
           .build();
     }
 
-    // TODO: [doug] 2020-06-18, Thu, 0:42 handle potential errors from authService
     String token;
     try {
-      token = authService.createToken(body.getUsername(), body.getPassword());
+      token = authService.createToken(credentials.getUsername(), credentials.getPassword());
+    } catch (UnauthorizedException e) {
+      return Response.status(Response.Status.UNAUTHORIZED)
+          .entity(new Error("Failed to create token: " + e.getMessage()))
```

Comment:
Is there any situation where `e.getMessage()` would have sensitive data that you wouldn't want to return back?

Reply:
No because it's just going to contain what the authService passes it. So as long as we don't throw any exceptions with sensitive data we're fine.